web/js/codeworld.js

```diff
@@ -158,6 +158,63 @@ function initCodeworld() {
     }
 }
 
+var cStream;
```

Comment:
I think it would be nicer if you encapsulated all of these global variables into a class.